### ISSUE TITLE
BUG: DOP853 with complex data computes complex error norm, causing step failure

### ISSUE THREAD

**Issue by zachjweiner** [scipy/scipy#11260]
I find that the new `DOP853` method for a problem in the complex domain raises a `TypeError` when trying to compute `min_step` [here](https://github.com/scipy/scipy/blob/master/scipy/integrate/_ivp/rk.py#L120). It seems to me that `DOP853._estimate_error_norm` (in contrast to `RungeKutta._estimate_error_norm`) may return a complex number, leading to a complex increment in `t`.

I suspect the L_2 norms computed in [here](https://github.com/scipy/scipy/blob/master/scipy/integrate/_ivp/rk.py#L500) just need to be computed with the same `norm` function that `RungeKutta` calls (the problem being that `abs` isn't called before squaring).

#### Reproducing code example:

```
import numpy as np
from scipy.integrate._ivp.rk import DOP853, RK45

def f(t, y):
    return - 3 * y

dop = DOP853(t0=0, t_bound=1, fun=f, y0=np.array([1j]))

print(dop._estimate_error_norm(dop.K, 1, np.ones(1)))
dop.step()
dop.step()
print(dop.t)
dop.step()
```

prints two complex numbers (for the error norm and for `t` after two timesteps) on my laptop. Calling `step` the third time leads to the error below, which does not occur upon replacing `DOP853` above with, say, `RK45`.

#### Error message:

```
TypeError                                 Traceback (most recent call last)
<ipython-input-123-accff485a0c9> in <module>
     13 dop.step()
     14 print(dop.t)
---> 15 dop.step()

~/miniconda3/lib/python3.7/site-packages/scipy/integrate/_ivp/base.py in step(self)
    180         else:
    181             t = self.t
--> 182             success, message = self._step_impl()
    183 
    184             if not success:

~/miniconda3/lib/python3.7/site-packages/scipy/integrate/_ivp/rk.py in _step_impl(self)
    118         atol = self.atol
    119 
--> 120         min_step = 10 * np.abs(np.nextafter(t, self.direction * np.inf) - t)
    121 
    122         if self.h_abs > max_step:

TypeError: ufunc 'nextafter' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
```

#### Scipy/Numpy/Python version information:

```
1.4.0 1.17.3 sys.version_info(major=3, minor=7, micro=3, releaselevel='final', serial=0)
```

**Comment by prof-danielc** [comment]
Also faced this error with the exact use case @zachjweiner indicated. This currently makes DOP853 unusable for complex data types.

**Comment by zachjweiner** [comment]
@prof-danielc, here's the simple fix I've been using:

```python
import numpy as np
from scipy.integrate._ivp.rk import DOP853 as DOP853_


class DOP853(DOP853_):
    def _estimate_error_norm(self, K, h, scale):
        err5 = np.dot(K.T, self.E5) / scale
        err3 = np.dot(K.T, self.E3) / scale

        # fixes bug in scipy 1.4.1
        from scipy.integrate._ivp.common import norm
        err5_norm_2 = norm(err5)
        err3_norm_2 = norm(err3)
        denom = err5_norm_2 + 0.01 * err3_norm_2
        return np.abs(h) * err5_norm_2 / np.sqrt(denom * len(scale))
```

Then just pass the class as the method kwarg: `solve_ivp(..., method=DOP853)`.

IIRC, this is the same norm function used by the other RK routines. I'll try and submit a PR for this soon.

**Comment by perlinm** [comment]
I think @zachjweiner's fix in https://github.com/scipy/scipy/issues/11260#issuecomment-674413860 is almost correct, but not quite.  If I take a look at `scipy.integrate._ivp.rk.DOP853._estimate_error_norm`:
```python
def _estimate_error_norm(self, K, h, scale):
    err5 = np.dot(K.T, self.E5) / scale
    err3 = np.dot(K.T, self.E3) / scale

    err5_norm_2 = np.sum(err5**2)
    err3_norm_2 = np.sum(err3**2)
    denom = err5_norm_2 + 0.01 * err3_norm_2
    return np.abs(h) * err5_norm_2 / np.sqrt(denom * len(scale))
```
and compare it with `scipy.integrate._ivp.common.norm`:
```python
def norm(x):
    """Compute RMS norm."""
    return np.linalg.norm(x) / x.size ** 0.5
```
it looks like `err5_norm_2` and `err3_norm_2` should be the squared Euclidean (i.e. L2) norms of `err5` and `err3`, rather than the RMS norms used in @zachjweiner's fix.  I think the correct fix to DOP853 should therefore be:
```python
import numpy as np
from scipy.integrate._ivp.rk import DOP853 as DOP853_

class DOP853(DOP853_):
    def _estimate_error_norm(self, K, h, scale):
        err5 = np.dot(K.T, self.E5) / scale
        err3 = np.dot(K.T, self.E3) / scale

        err5_norm_2 = np.sum(abs(err5)**2)
        err3_norm_2 = np.sum(abs(err3)**2)
        denom = err5_norm_2 + 0.01 * err3_norm_2
        return np.abs(h) * err5_norm_2 / np.sqrt(denom * len(scale))
```
which, as in @zachjweiner's case, should be passed as a kwarg to `scipy.integrate.solve_ivp(..., method = DOP853)`.
